coalahtml/_coalahtml/app/controllers/main.js

```diff
@@ -2,45 +2,47 @@
 
 angular.module('coalaHtmlApp')
   .controller('MainCtrl', function ($scope, $http) {
+    $http.get("data/Constants.json").success(function(content) {
+      $scope.Constants = content;
+      parseCoalaProject($scope.Constants.data);
+    });
     var parseCoalaProject = function(projectDir, coalaJSON) {
-      coalaJSON = coalaJSON || projectDir + '/coala.json';
-      $scope.data = [];
-      $http.get(coalaJSON).success(function(response){
-        var knownFiles = {};
-        $scope.content = {};
-
-        var parseResult = function(el){
-          el.affected_code.forEach(function(sourceRange){
-              if (!(sourceRange.file in knownFiles)){
-                  knownFiles[sourceRange.file] = [];
-
-                  $http.get(projectDir + "/" + sourceRange.file)
-                    .success(function(content) {
-                      $scope.content[sourceRange.file] = content.split("\n");
+      $http.get(projectDir + $scope.Constants.file_data)
+        .success(function(file_data) {
+            $scope.file_dict = {};
+            Object.keys(file_data).forEach(function(file_name) {
+              if (!(file_name in $scope.file_dict)) {
+                $scope.file_dict[file_name]=file_data[file_name];
+              }
+            });
+          coalaJSON = coalaJSON || projectDir + $scope.Constants.coala;
+          $scope.data = [];
+          $http.get(coalaJSON).success(function(response) {
+            var knownFiles = {};
+            var parseResult = function(el) {
+                el.affected_code.forEach(function(sourceRange){
+                    if (!(sourceRange.file in knownFiles)){
+                        knownFiles[sourceRange.file] = [];
+                    }
+                    knownFiles[sourceRange.file].push({
+                      "start":    sourceRange.start.line,
+                      "end":      sourceRange.end.line,
+                      "diffs":    el.diffs,
+                      "message":  el.message,
+                      "origin":   el.origin,
+                      "severity": el.severity
                     });
+                });
+            };
+            for (var section in response.results) {
+              if (response.results[section][0]){ // If error exists
```

Comment:
Hm makes sense. Sure. Making changes. Thanks :+1:
ᐧ

On Mon, May 23, 2016 at 10:37 AM, Attila <notifications@github.com> wrote:

> In coalahtml/_coalahtml/app/controllers/main.js
> <https://github.com/coala-analyzer/coala-html/pull/46#discussion_r64169870>
> :
>
> >                      });
> > +                });
> > +            };
> > +            for (var section in response.results) {
> > +              if (response.results[section][0]){ // If error exists
>
> Checking the length would be more transparent and correct, because in the
> array you can have one element which is False.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/coala-analyzer/coala-html/pull/46/files/303f23533e52f0722dec3ad0291658c8adffe5fb..7fafb94dc8226dd32fc0621e3fffcb23b9c4728b#r64169870>
>